scripts/modelscope-text2vid.py

```diff
@@ -140,6 +143,61 @@ def process(skip_video_creation, ffmpeg_location, ffmpeg_crf, ffmpeg_preset, fps
             #latents should have shape num_sample, 4, max_frames, latent_h,latent_w
             print("Computing latents")
             latents = pipe.compute_latents(vd_out).to(device)
+        
+        elif do_inpainting:
+            images=[]
+            print("gir",inpainting_image)
+            print(inpainting_image.name)
+            for i in range(frames):
+                image=Image.open(inpainting_image.name).convert("RGB")
+                image=image.resize((width,height), Image.ANTIALIAS)
+                array = np.array(image)
+                images+=[array]
+
+            images=np.stack(images)# f h w c
+            batches=1
+            n_images=np.tile(images[np.newaxis, ...], (batches, 1, 1, 1, 1)) # n f h w c
+            bcfhw=n_images.transpose(0,4,1,2,3)
+            #convert to 0-1 float
+            bcfhw=bcfhw.astype(np.float32)/255
+            bfchw=bcfhw.transpose(0,2,1,3,4)#b c f h w
+
+            print(f"Converted the frames to tensor {bfchw.shape}")
+
+            vd_out=torch.from_numpy(bcfhw).to("cuda")
+
+            #should be -1,1, not 0,1
+            vd_out=2*vd_out-1
+
+            #latents should have shape num_sample, 4, max_frames, latent_h,latent_w
+            #but right now they have shape num_sample=1,4, 1 (only used 1 img), latent_h, latent_w
+            print("Computing latents")
+            image_latents = pipe.compute_latents(vd_out).numpy()
+            #padding_width = [(0, 0), (0, 0), (0, frames-inpainting_frames), (0, 0), (0, 0)]
+            #padded_latents = np.pad(image_latents, pad_width=padding_width, mode='constant', constant_values=0)
+
+            latent_h=height//8
+            latent_w=width//8
+            latent_noise=np.random.normal(size=(1,4,frames,latent_h,latent_w))
+            mask=np.ones(shape=(1,4,frames,latent_h,latent_w))
+
+            if len(inpainting_weights)>0:
+                mask_weights=eval(inpainting_weights)
```

Comment:
Please, don't use `eval`. It's absolutely unsafe. If you have to specify the weights dynamically, take a looks how it's done in Deforum with a more safe library `numexpr` https://github.com/deforum-art/deforum-for-automatic1111-webui/blob/automatic1111-webui/scripts/deforum_helpers/animation_key_frames.py

Reply:
can I just use safeeval (https://github.com/viur-framework/safeeval) or just put a warning that says "don't paste untrusted code into this box"?